python_modules/dagster/dagster/_core/event_api.py

```diff
@@ -146,3 +146,155 @@ def __new__(
+            ),
+            after_timestamp=check.opt_float_param(after_timestamp, "after_timestamp"),
+            before_timestamp=check.opt_float_param(before_timestamp, "before_timestamp"),
+            storage_ids=check.opt_nullable_sequence_param(storage_ids, "storage_ids", of_type=int),
+            tags=check.opt_nullable_mapping_param(tags, "tags", key_type=str),
+        )
+
+    def to_event_records_filter(self, event_type: DagsterEventType) -> EventRecordsFilter:
+        return EventRecordsFilter(
+            event_type=event_type,
+            asset_key=self.asset_key,
+            asset_partitions=self.asset_partitions,
+            after_cursor=self.after_cursor,
+            before_cursor=self.before_cursor,
+            after_timestamp=self.after_timestamp,
+            before_timestamp=self.before_timestamp,
+            storage_ids=self.storage_ids,
+            tags=self.tags,
+        )
+
+
+@whitelist_for_serdes
+class RunStatusEventRecordsFilter(
+    NamedTuple(
+        "_RunStatusEventRecordsFilter",
+        [
+            ("event_type", PublicAttr[DagsterEventType]),
+            ("after_cursor", PublicAttr[Optional[Union[int, RunShardedEventsCursor]]]),
+            ("before_cursor", PublicAttr[Optional[Union[int, RunShardedEventsCursor]]]),
+            ("after_timestamp", PublicAttr[Optional[float]]),
+            ("before_timestamp", PublicAttr[Optional[float]]),
+            ("storage_ids", Optional[Sequence[int]]),
+        ],
+    )
+):
+    """Defines a set of filter fields for fetching a set of asset event records.
+
+    Args:
+        event_type (DagsterEventType): Filter argument for dagster event type
+        after_cursor (Optional[Union[int, RunShardedEventsCursor]]): Filter parameter such that only
+            records with storage_id greater than the provided value are returned. Using a
+            run-sharded events cursor will result in a significant performance gain when run against
+            the sqlite storage implementation (which is run-sharded).
+        before_cursor (Optional[Union[int, RunShardedEventsCursor]]): Filter parameter such that
+            records with storage_id less than the provided value are returned. Using a run-sharded
+            events cursor will result in a significant performance gain when run against the sqlite
+            storage implementation (which is run-sharded).
+        after_timestamp (Optional[float]): Filter parameter such that only event records for
+            events with timestamp greater than the provided value are returned.
+        before_timestamp (Optional[float]): Filter parameter such that only event records for
+            events with timestamp less than the provided value are returned.
+        storage_ids (Optional[Sequence[int]]): Filter parameter such that only event records for
+            the given storage ids are returned.
+    """
+
+    def __new__(
+        cls,
+        event_type: DagsterEventType,
```

Comment:
probably worth run time enforcing the specific set of dagster event types we think this should support. The type signature maybe as well